tests/kafkatest/services/connect.py

```diff
@@ -432,9 +482,14 @@ def start_node(self, node, **kwargs):
             self.start_and_return_immediately(node, 'distributed', '')
         elif self.startup_mode == self.STARTUP_MODE_LISTEN:
             self.start_and_wait_to_start_listening(node, 'distributed', '')
-        else:
-            # The default mode is to wait until the complete startup of the worker
+        elif self.startup_mode == self.STARTUP_MODE_JOIN:
             self.start_and_wait_to_join_group(node, 'distributed', '')
+        elif self.startup_mode is None or self.startup_mode == self.STARTUP_MODE_HEALTH_CHECK:
```

Comment:
nit: This start_node method is pretty similar in both ConnectDistributedService and ConnectStandaloneService, maybe the common parts should be in the ConnectServiceBase with a template method for the connector_config_templates processing